cardio_crumble.opam

```diff
@@ -10,11 +10,12 @@ homepage: "https://github.com/username/reponame"
 doc: "https://url/to/documentation"
 bug-reports: "https://github.com/username/reponame/issues"
 depends: [
-  "ocaml"
-  "dune" {>= "3.3"}
+  "ocaml" {>= "5.0.0"}
+  "dune" {>= "3.3" & >= "3.7.0"}
+  "portmidi" {= "0.1"}
+  "cmdliner" {= "1.1.1"}
+  "ocamlformat" {>= "0.24.1"}
```

Comment:
Adding `ocamlformat` to the dependencies does indeed seem like a good idea now that you've found out that all devs need the same `ocamlformat` version.

This is a bit of a complicated topic though. If you list a dependency without specifying any variable such as `with-test` or `with-dev-setup`, those dependencies are considered hard dependencies, i.e. they will be installed for everyone who wants to use `cardio-crumble`. However, `ocamlformat` is only necessary for people developing `ocamlformat`.

In fact, this topic is even more complicated due to the way how `opam` installs tools (such as `ocamlformat`). I could give more details, if you're really interested. The summary is: I'd advice not to add `ocamlformat` to the dependencies at all.

Reply:
For now, I'll remove ocamlformat as a dependency @pitag-ha !
But yes, I'd like to know more on this topic, can you please guide me on this (I don't have much knowledge about handling big projects with modules at the moment).

Reply:
> For now, I'll remove ocamlformat as a dependency @pitag-ha !

Thanks!

> I'd like to know more on this topic, can you please guide me on this

Do you have a concrete question you'd like me to answer?

Reply:
No, you answered my doubts with the comment below!
I just wanted to know the reasoning behind the versions, which is clear to me now. Thanks @pitag-ha !!